rtl/eclrtl/eclhelper_dyn.cpp

```diff
@@ -0,0 +1,166 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "platform.h"
+#include "jptree.hpp"
+#include "eclrtl.hpp"
+#include "eclhelper.hpp"
+#include "rtlds_imp.hpp"
+#include "eclhelper_base.hpp"
+#include "eclhelper_dyn.hpp"
+
+#include "rtlfield.hpp"
+#include "rtlrecord.hpp"
+#include "rtldynfield.hpp"
+#include "rtlkey.hpp"
+
+//---------------------------------------------------------------------------
+static const RtlRecordTypeInfo &loadTypeInfo(IPropertyTree &xgmml, IRtlFieldTypeDeserializer *deserializer, const char *key)
+{
+    StringBuffer xpath;
+    MemoryBuffer binInfo;
+    xgmml.getPropBin(xpath.setf("att[@name='%s_binary']/value", key), binInfo);
+    assertex(binInfo.length());
+    const RtlTypeInfo *typeInfo = deserializer->deserialize(binInfo);
+    assertex(typeInfo && typeInfo->getType()==type_record);
+    return *(RtlRecordTypeInfo *) typeInfo;
+}
+
+class ECLRTL_API CDynamicDiskReadArg : public CThorDiskReadArg
+{
+public:
+    CDynamicDiskReadArg(IPropertyTree &_xgmml) : xgmml(_xgmml)
+    {
+        indeserializer.setown(createRtlFieldTypeDeserializer());
+        outdeserializer.setown(createRtlFieldTypeDeserializer());
+        in.setown(new CDynamicOutputMetaData(loadTypeInfo(xgmml, indeserializer, "input")));
+        out.setown(new CDynamicOutputMetaData(loadTypeInfo(xgmml, outdeserializer, "output")));
+        inrec = &in->queryRecordAccessor(true);
+        numOffsets = inrec->getNumVarFields() + 1;
+        translator.setown(createRecordTranslator(queryOutputMeta()->queryRecordAccessor(true), *inrec));
+        if (xgmml.hasProp("att[@name=\"keyfilter\"]"))
+            flags |= TDRkeyed;
+    }
+    virtual bool needTransform() override
+    {
+        return true;
+        //return translator->needsTranslate();
```

Comment:
should this be removed, or does this case need more work.

Reply:
removed